src/e2e-test/java/io/cdap/plugin/common/stepsdesign/GCS.java

```diff
@@ -0,0 +1,131 @@
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+import io.cdap.plugin.gcp.common.SourceOutputFormatProvider;
+import io.cucumber.java.en.Then;
+
+import java.io.FileReader;
+import java.io.IOException;
+
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.StorageException;
+import io.cdap.e2e.utils.CdfHelper;
+import io.cdap.e2e.utils.PluginPropertyUtils;
+import io.cdap.e2e.utils.StorageClient;
+import io.cucumber.java.en.Then;
+import kafka.utils.Json;
+import org.apache.directory.api.util.Strings;
+import org.apache.spark.sql.execution.columnar.NULL;
+import org.junit.Assert;
+import stepsdesign.BeforeActions;
+
+import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class GCS {
+  private static final Gson gson = new Gson();
+
+  public static void main(String[] args) {
+    validateOutputFileGeneratedByGCSSinkPluginIsEqualToExpectedOutputFile();
+  }
+
+  public static boolean validateJsonArray(JsonArray v1, JsonArray v2) {
+    if(!(v1.size() == v1.size())) return false;
+    for (int i =0 ; i < v1.size(); i++ ) {
+      if(!validateJsonElement(v1.get(i),v2.get(i))) return false;
+    }
+    return  true;
+  }
+
+  public static boolean validateJsonPrimitive(JsonElement v1, JsonElement v2) {
+    // TODO Update this later to a different function
+    return v1.getAsString().equals(v2.getAsString());
+  }
+
+  public static boolean validateJsonElement(JsonElement v1, JsonElement v2) {
+    // Taking v1 as source of truth
+    if (v1.isJsonObject()) {
+      // Base case of checking v2
+      if (!v2.isJsonObject()) return false;
+      return validateJsonObject(v1.getAsJsonObject(), v2.getAsJsonObject());
+
+    } else if (v1.isJsonArray()) {
+      if (!v2.isJsonArray()) return false;
+      return validateJsonArray(v1.getAsJsonArray(),v2.getAsJsonArray());
+
+    } else if (v1.isJsonPrimitive()) {
+      if (!v2.isJsonPrimitive()) return false;
+      return validateJsonPrimitive(v1, v2);
+
+    } else if (v1.isJsonNull()) {
+      if (!v2.isJsonNull()) return false;
+      return true;
+    }
+    return false;
+  }
+  public static boolean validateJsonObject(JsonObject v1, JsonObject v2) {
+    Set<Map.Entry<String, JsonElement>> v1EntrySet = v1.entrySet();
+    for (Map.Entry<String, JsonElement> entry : v1EntrySet) {
+      // Key
+      String key = entry.getKey();
+      // Values
+      JsonElement fileValue = v1.get(key);
```

Comment:
Why you are getting the key again in fileValue? Change this to 
JsonElement fileValue = entry.getValue();     
JsonElement gcsValue = v2.get(key);

Reply:
done
commit id- https://github.com/cloudsufi/google-cloud/commit/defc54df110255ef539dea8d206481cc145b2fec